test/component/storage/storage.go

```diff
@@ -1077,6 +1078,1407 @@ func TestStoreQuery(t *testing.T, provider spi.Provider) { // nolint: funlen //
 	})
 }
 
+// TestStoreQueryWithSortingAndInitialPageOptions tests common Store Query functionality when the sorting and initial
```

Comment:
In the future I should probably update this to use some sort of test table in order to cut down on the number of `t.Run(...` I have